lib/core/test/unit/Cardano/Wallet/DB/SqliteFileModeSpec.hs

```diff
@@ -0,0 +1,284 @@
+import qualified Data.Set as Set
+
+spec :: Spec
+spec =  do
+    describe "Check db opening/closing" $ do
+
+        it "opening and closing of db works" $ do
+            replicateM_ 25 openCloseDB
+
+    before (fileDBLayer >>= dbCleaned)$
+        describe "Check db reading/writing from/to file and cleaning" $ do
+
+        it "create and list wallet works" $ \(conn, db) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            listWallets db `shouldReturn` [testPk]
+            close conn
+            (conn1, db1) <- fileDBLayer
+            runExceptT (createWallet db1 testPk testCp testMetadata)
+                `shouldReturn` (Left (ErrWalletAlreadyExists testWid))
+            close conn1
+            ( testOpeningCleaning
+                listWallets
+                [testPk]
+                [] )
+
+        it "create and get meta works" $ \(conn, db) -> do
+            now <- getCurrentTime
+            let md = testMetadata { passphraseInfo = Just $ WalletPassphraseInfo now }
+            unsafeRunExceptT $ createWallet db testPk testCp md
+            readWalletMeta db testPk `shouldReturn` Just md
+            close conn
+            ( testOpeningCleaning
+                (`readWalletMeta` testPk)
+                (Just md)
+                Nothing )
+
+        it "create and get private key" $ \(conn,db) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            readPrivateKey db testPk `shouldReturn` Nothing
+            let Right phr = fromText "simplephrase"
+                k = unsafeGenerateKeyFromSeed (coerce phr, coerce phr) phr
+            h <- encryptPassphrase phr
+            unsafeRunExceptT (putPrivateKey db testPk (k, h))
+            readPrivateKey db testPk `shouldReturn` Just (k, h)
+            close conn
+            ( testOpeningCleaning
+                (`readPrivateKey` testPk)
+                (Just (k, h))
+                Nothing )
+
+        it "put and read tx history" $ \(conn,db) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            runExceptT (putTxHistory db testPk testTxs) `shouldReturn` Right ()
+            readTxHistory db testPk `shouldReturn` testTxs
+            close conn
+            ( testOpeningCleaning
+                (`readTxHistory` testPk)
+                testTxs
+                Map.empty )
+
+        it "put and read checkpoint" $ \(conn,db) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            runExceptT (putCheckpoint db testPk testCp) `shouldReturn` Right ()
+            readCheckpoint db testPk `shouldReturn` Just testCp
+            close conn
+            ( testOpeningCleaning
+                (`readCheckpoint` testPk)
+                (Just testCp)
+                Nothing )
+
+    withDB inMemoryDBLayer $
+        describe "random operation chunks property when writing to/reading from file" $ do
```

Comment:
I'd suggest the following:

```hs
    withDB inMemoryDBLayer $
        describe
            "random operation chunks property when writing to/reading from \
            \file" $ do
```

For compliance with https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#limit-line-length-to-80-characters